src/suspense.js

```diff
@@ -0,0 +1,56 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+// having a "custom class" here saves 50bytes gzipped
+export function s(props) {}
+s.prototype = new Component();
+
+/**
+ * @param {Promise} promise The thrown promise
+ */
+s.prototype._childDidSuspend = function(promise) {
+	this.setState({ _loading: true });
+	const cb = () => { this.setState({ _loading: false }); };
+
+	// Suspense ignores errors thrown in Promises as this should be handled by user land code
+	promise.then(cb, cb);
+};
+
+s.prototype.render = function(props, state) {
+	return state._loading ? props.fallback : props.children;
+};
+
+// exporting s as Suspense instead of naming the class iself Suspense saves 4 bytes gzipped
+// TODO: does this add the need of a displayName?
+export const Suspense = s;
+
+export function lazy(loader) {
+	let prom;
+	let component;
+	let error;
+
+	function Lazy(props) {
+		if (!prom) {
+			prom = loader();
+			prom.then(
+				(exports) => {
+					component = exports.default;
+					Lazy.displayName = 'Lazy(' + (component.displayName || component.name) + ')';
```

Comment:
Just checked against the experimental react-devtools to see how they detect both `Suspense` and `Lazy` components. Turns out that the `displayName` for `Lazy` doesn't need to be aware of the component it is rendering. It's printed like a standard component inside the tree.

```txt
<Suspense>
  <Lazy>
    <MyComponent>
```

With this in mind just doing `Lazy.displayName = 'Lazy'` is enough. That can happen outside the declaration itself.

In the long run we'll probably switch to just flagging any special components with a constant instead of a string. That way we could detect `Provider` and `Consumer` components too. The devtools render these slightly differently, but I haven't gotten around to implement that yet. This is out of scope for this PR though and the above solution is enough for now :+1: